### PR TITLE
Sinusoidal dataset

### PR DESCRIPTION
**Pull request by pablomm**
Added maker of a sinusoidal process, used in a example of registration.

### CODE REVIEW

File: fda/datasets/_samples_generators.py [hunk 1]
```diff
@@ -49,3 +49,57 @@ def make_gaussian_process(n_samples: int=100, n_features: int=100, *,
...
+        amplitude_std: Standard deviation of the amplitude.
+        error_std: Standard deviation of the gaussian Noise.
+        random_state: Random state.
+
+    Returns:
+        :class:`FDataGrid` object comprising all the samples.
+
+    """
+
+    sklearn.utils.check_random_state(random_state)
```
Comment (vnmabus) [inline fda/datasets/_samples_generators.py]: You should assign the result.

File: fda/datasets/_samples_generators.py [hunk 2]
```diff
@@ -49,3 +49,57 @@ def make_gaussian_process(n_samples: int=100, n_features: int=100, *,
...
+
+    t = np.linspace(start, stop, n_features)
+
+    alpha = np.diag(np.random.normal(amplitude_mean, amplitude_std, n_samples))
+
+    phi = np.outer(np.random.normal(phase_mean, phase_std,  n_samples),
+                   np.ones(n_features))
+
+    error = np.random.normal(0, error_std, (n_samples, n_features))
```
Comment (vnmabus) [inline fda/datasets/_samples_generators.py]: You should use the random state.

File: fda/datasets/_samples_generators.py [hunk 3]
```diff
@@ -49,3 +49,57 @@ def make_gaussian_process(n_samples: int=100, n_features: int=100, *,
...
+    Returns:
+        :class:`FDataGrid` object comprising all the samples.
+
+    """
+
+    sklearn.utils.check_random_state(random_state)
+
+    t = np.linspace(start, stop, n_features)
+
+    alpha = np.diag(np.random.normal(amplitude_mean, amplitude_std, n_samples))
```
Comment (vnmabus) [inline fda/datasets/_samples_generators.py]: You should use the random state.

File: fda/datasets/_samples_generators.py [hunk 4]
```diff
@@ -49,3 +49,57 @@ def make_gaussian_process(n_samples: int=100, n_features: int=100, *,
...
+
+    phi = np.outer(np.random.normal(phase_mean, phase_std,  n_samples),
```
Comment (vnmabus) [inline fda/datasets/_samples_generators.py]: You should use the random state.

File: fda/datasets/_samples_generators.py (hunk 1 above)
Comment (pablomm) [reply fda/datasets/_samples_generators.py]: I thought it works like seed, thanks for the correction. Fixed in e72bf90.

File: fda/datasets/_samples_generators.py (hunk 2 above)
Comment (pablomm) [reply fda/datasets/_samples_generators.py]: Fixed in e72bf90.

File: fda/datasets/_samples_generators.py (hunk 3 above)
Comment (pablomm) [reply fda/datasets/_samples_generators.py]: Fixed in e72bf90.

File: fda/datasets/_samples_generators.py (hunk 4 above)
Comment (pablomm) [reply fda/datasets/_samples_generators.py]: Fixed in e72bf90.
